contractdata/AMBROSE/_AMBROSE_MASTERY.json

```diff
@@ -0,0 +1,598 @@
+[
```

Comment:
The completion data is reset and includes the MaxLevel - this is not always 20, but sometimes 5 and in case of evergreen 25 and in the full release even more.

The reason I have the format as is, is because it significantly simplifies the loading of the data into the proper locations. Otherwise you will either end up:
- Generating this exact output when loading the .csp file with the data (startup overhead, no memory difference)
- Generating this data every time you need it (aka: mapping - no startup overhead, less static memory)

So it's a give or take situation. From a maintainability perspective a smaller revised format is easier, but it comes at a cost.

Reply:
Adding onto this, `TypeLocaKey` can be generated from the unlockable, so here's an updated format:
```json
{
    "Id": "LOCATION_PARENT_ROCKY",
    "Levels": [
        {
            "Level": 2,
            "Unlockable": "AGENCYPICKUP_ROCKY_SMALL_BIGCENTRALTREE",
            "IsLevelMarker": false
        }
    ]
}
```

Reply:
> The completion data is reset and includes the MaxLevel
What do you measn by "reset", and "MaxLevel" can be added by a property, just like Id.

I'm sure @RDIL can comment about the generation part, I think having the smaller format is more beneficial as we're not storing the _exact_ same data twice, and it's actually better for maintaining it as if the unlockable changes between updates (i.e. a suit or weapon), we'd have to update the mastery json too while we'd normally just update the one `allunlockables`.

Reply:
Unless the duplication has a very good reason for being present, I'd like to avoid it.  Very likely, reducing memory by, say, 10 megabytes, is a better improvement then saving less then say 50 milliseconds.

Reply:
It also isn't something that would be ran often so it wouldn't impact much.

Reply:
> What do you mean by "reset", and "MaxLevel" can be added by a property, just like Id.
The MaxLevel determines the maximum level of the mastery without having the scan through the mastery data to grab the highest available level.

But I understand where you guys want to go. Runtime generation of the full data on-demand it is.